packages/core-data/src/entities.js

```diff
@@ -293,75 +293,99 @@ async function loadPostTypeEntities() {
 	const postTypes = await apiFetch( {
 		path: '/wp/v2/types?context=view',
 	} );
-	return Object.entries( postTypes ?? {} ).map( ( [ name, postType ] ) => {
-		const isTemplate = [ 'wp_template', 'wp_template_part' ].includes(
-			name
-		);
-		const namespace = postType?.rest_namespace ?? 'wp/v2';
-		return {
-			kind: 'postType',
-			baseURL: `/${ namespace }/${ postType.rest_base }`,
-			baseURLParams: { context: 'edit' },
-			name,
-			label: postType.name,
-			transientEdits: {
-				blocks: true,
-				selection: true,
-			},
-			mergedEdits: { meta: true },
-			rawAttributes: POST_RAW_ATTRIBUTES,
-			getTitle: ( record ) =>
-				record?.title?.rendered ||
-				record?.title ||
-				( isTemplate
-					? capitalCase( record.slug ?? '' )
-					: String( record.id ) ),
-			__unstablePrePersist: isTemplate ? undefined : prePersistPostType,
-			__unstable_rest_base: postType.rest_base,
-			syncConfig: {
-				fetch: async ( id ) => {
-					return apiFetch( {
-						path: `/${ namespace }/${ postType.rest_base }/${ id }?context=edit`,
-					} );
+	const entities = Object.entries( postTypes ?? {} ).map(
+		async ( [ name, postType ] ) => {
+			const isTemplate = [ 'wp_template', 'wp_template_part' ].includes(
+				name
+			);
+			const namespace = postType?.rest_namespace ?? 'wp/v2';
+			// If meta is not present, fetch it.
+			const registeredMeta =
+				postType.meta ||
+				(
+					await apiFetch( {
+						path: '/wp/v2/' + postType?.rest_base + '?context=edit',
+						method: 'OPTIONS',
+					} )
+				)?.schema?.properties?.meta?.properties;
```

Comment:
It seems `postType.meta` is usually undefined. It feels too much to make this extra call per post type.

Reply:
We could move this API call to the `post-meta` getFieldsList function to only do the fetch when necessary.